File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;
class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> map = new HashMap<String, Integer>();
        for (String name : completion) {
            int value = map.get(name) == null ? 1 : map.get(name) + 1;
            map.put(name, value);
        }
        String answer = "";
        for (String name : participant) {
            if (!map.containsKey(name) || map.get(name) == 0) {
                answer = name;
            } else {
                map.replace(name, map.get(name) - 1);
            }
        }

        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    h = {}
    for c in completion:
        if c in h:
            h[c] += 1
        else:
            h[c] = 1
    for p in participant:
        if p not in h or h[p] == 0:
            return p
        else:
            h[p] -= 1
            